ecs-agent/api/ecs/client/ecs_client.go

```diff
@@ -776,6 +782,7 @@ func (client *ecsClient) discoverPollEndpoint(containerInstanceArn string,
 		}
 		return nil, err
 	}
+	client.metricsFactory.New(metrics.DiscoverPollEndpointDurationName).WithGauge(time.Since(discoverPollEndpointStartTime)).Done(nil)
```

Comment:
Besides this gauge metric, can we emit one more counter metric to monitor if the discoverPollEndpoint call is successful or not?

Reply:
Sounds good, think this will be a good addition to get more insights into `DiscoverPollEndpoint()`